jms/1.8/modules/ROOT/pages/jms-troubleshooting.adoc

```diff
@@ -0,0 +1,132 @@
+You can configure the session acknowledgment for an application as:
+
+* *IMMEDIATE*
+* *AUTO*
+* *MANUAL*
+* *DUPS_OK*
+
+When you set the session acknowledgment to *IMMEDIATE*, the message gets acknowledged as soon as it's received. Do not mix *IMMEDIATE* acknowledgment with *Ack* or *Recovery session* operations.
+
+When you set the session acknowledgment to *AUTO*, the *On New Message* source takes control of the acknowledgment when the event finishes, or the session recovers when an error is thrown while processing the event. Adding a manual acknowledgment or session recover in an application with *AUTO* acknowledgment leads to duplicate operations on a channel.
+
+When you set the session acknowledgment to either *MANUAL* or *DUPS_OK*, check that every path of the flow execution gets a message acknowledgment or a session recover. Otherwise, a message can leak and lead to an unresponsive consumer when the max depth of the unit of work is reached.
+
+To resolve this issue:
+
+Take a detailed look at the flow design and be sure that every message is acknowledged or the session recovers.
+
+== Troubleshoot JMS Stopped Consuming Messages
+
+Usually when JMS stopped consuming messages is related to a large number of unacknowledged messages. When your  Mule app that has an *On New Message* source or *Consume* operation with manual acknowledgement of messages, and not every path of the message has the proper acknowledgement or session recovery configuration, the Mule app can lead to an unresponsive consumer.
+
+To resolve this error:
+
+Check the depth of unacknowledged messages on the broker side and review the application logic.
+
+== Troubleshoot Redeployment Errors When Using Binding Mode
+
+When using binding mode for connections, the driver loads a JNI resource. Because the JNI resource can be loaded only once, when the application is redeployed, an error is thrown.
+
+To resolve this error:
+
+Add the driver dependency at the domain level, which enables application redeployment.
+
+== HA Enviroments and Durable Suscriptions
+
+The configuration of HA systems consists of two components: the primary node (which is providing continuous service) and the secondary node (which is a clone of the previous one so that it provides the same service). Both of them must be identical so that the end-user is not aware of the system crash and can continue to use it transparently.
+When you are working with Durable Suscriptions, only one node can getting the connection and the other one fails.
+In this kind or scenarios, the nodes are not clones since one of them has something that the other does not: the connection.
+The possibility of using connections that cannot be initialized in multiple nodes simultaneously was not considered in the design of Mule running in HA clusters. The current design is aimed at having all nodes initializing connections to ensure that they are capable of facing a failure of the primary node.
```

Comment:
I'd reword this. Saying that a feature was not considered give's me a negative sentiment.